File: dsh/identity/invariant.py
"""
Package-owned invariant companion for dsh.identity.
1:1 with reference @deepseek-ai/dsh-anonymous-user-id/invariant.
Python 3.8.10 compatible.
"""

from typing import Any, Callable, Optional

PACKAGE_NAME = "@deepseek-ai/dsh-anonymous-user-id"
name = "anonymous-user-id-invariant"
inject = ["invariants"]


def install(ctx: Any, fail: Callable[[str], None]) -> None:
    pass


def apply(ctx: Any) -> Optional[Any]:
    if hasattr(ctx, "invariants") and hasattr(ctx.invariants, "register"):
        return ctx.invariants.register(PACKAGE_NAME, install)
    return None

File: packages/experimental/webworker-runtime/lib/invariant.js
//#region lib/types/invariant.js
/**
* Package-owned invariant companion for `@deepseek-ai/dsh-experimental-webworker-runtime`.
* @module @deepseek-ai/dsh-experimental-webworker-runtime/invariant
*/
const PACKAGE_NAME = "@deepseek-ai/dsh-experimental-webworker-runtime";
/** Cordis companion plugin name. */
const name = "webworker-runtime-invariant";
/** Service required before the companion can reserve package ownership. */
const inject = ["invariants"];
/**
* No runtime invariant: this package is pre-Cordis platform glue —
* the tree it boots runs the product packages' own invariants, and the
* assembly's contracts (image contract gate, tunnel refusals) fail loud at
* boot rather than drifting at run time.
*/
const install = () => {};
/**
* Register this package's invariant companion.
* @param ctx - Cordis context carrying the invariant service.
* @returns the installed registration's disposer after setup succeeds.
*/
const apply = (ctx) => Promise.resolve(ctx.invariants.register(PACKAGE_NAME, install));
//#endregion
export { apply, inject, name };
